pycqed/analysis_v2/base_analysis.py

```diff
@@ -199,9 +204,39 @@ def process_data(self):
     def prepare_plots(self):
         pass
 
-    def run_fitting(self):
+    def prepare_fitting(self):
         pass
 
+    def run_fitting(self):
+        # This function does the fitting and saving of the parameters
+        # based on the fit_dict options.
+        # Only model fitting is implemented yet. Minimizing fitting should
+        # be implemented here.
+        fit_guess_fn = self.fit_dict['fit_guess_fn']
+        fit_fn = self.fit_dict['fit_fn']
+        fit_yvals = self.fit_dict['fit_yvals']
+        fit_xvals = self.fit_dict['fit_xvals']
+        model = lmfit.Model(fit_fn)
+        if self.do_timestamp_blocks:
+            self.fit_dict['fit_results'] = []
+            for tt in range(len(fit_yvals)):
+                guess_dict = fit_guess_fn(fit_yvals[tt],
+                                          **fit_xvals[tt])
+                for key, val in list(guess_dict.items()):
+                    model.set_param_hint(key, **val)
+                params = model.make_params()
+                self.fit_dict['fit_results'].append(model.fit(fit_yvals[tt],
```

Comment:
I'm confused that the guess happens here (in the prepare fitting method) I would expect this to happen in the actual fitting method. Especially as it is usally the piece of code that takes most space. 

Otherwise the dictionary doesn't really add much in respect of enforcing structure.